src/celeste/twoStackQueue.js

```diff
@@ -0,0 +1,60 @@
+// Two Stack Queue (medium)
+
+// Implement a Queue using two stacks as defined below. You can assume that
+// the stacks implement push and pop in O(1).
+//
+// Your queue only needs enqueue and a dequeue methods and you should optimize
+// for the time cost of x calls on your queue.
+//
+// Your queue cannot be a class (it should hold no values, only the two stacks
+// should hold your data).
+// -----------------------------------------------------------------------------
+
+class Stack {
+  constructor(){
+    this.stack = []
+  }
+
+  pushIt(item){
+    this.stack.push(item)
+  }
+
+  popIt(){
+    return this.stack.pop()
+  }
+
+}
+
+const firstStack = new Stack()
+const secondStack = new Stack()
+// -----------------------------------------------------------------------------
+
+function enqueue(item) {
+  firstStack.pushIt(item)
+}
+
+function dequeue() {
+  let current = firstStack.popIt()
+  while (current) {
+    secondStack.pushIt(current)
+    current = firstStack.popIt()
+  }
+  let result = secondStack.popIt()
+  let backToFirst = secondStack.popIt()
+  while (backToFirst) {
+    firstStack.pushIt(backToFirst)
+    backToFirst = secondStack.popIt()
+  }
+  return result
+}
```

Comment:
This works but I think it's working a little too hard. 
For instance, if there's only one item in the firstStack and none in the second, you are moving the item to the second, then popping it off.

Also, there is never an instance in which dequeue could be O(1)

What if you checked stackTwo (which should be sorted in queue like order if you pop one by one all the items of stack one on to stack two), and if it's not empty just pop the top value which would be the next in queue and O(1) .

Then if stackTwo (aka the queue like behaving stack) is empty, pop and push each item of stack one in to stack two, and now stackTwo is once again your queue.

This way you only even look at stackOne if stackTwo is empty, you re-stack in to a queue, and until stackTwo is fully dequeued, all your dequeue() calls will be in O(1).